Add a newsletter Archetype

gmantele opened the pull request:
This PullRequest aims to help the newsletter editors. It adds an Archetype for a newsletter ; in other words, it adds a template file for a newsletter.

To create a new newsletter using this template, just run: `./hugo-bin/hugo new content newsletter/XXX.md`.
Then, just edit the resulting file `content/newsletter/XXX.md` and especially all content placeholders spread all over this file.

I have added some instructions in `CONTRIBUTING.md`. I hope it will be useful. And of course, if information are missing there, or if you think it should written somewhere else, I can easily change that.

What do you think about this solution? @JeremyMcCormick ? Deborah ?

JeremyMcCormick commented:
Thank you. This seems helpful!

I will try it out and let you know what I think...

JeremyMcCormick commented:
FYI, there is an open issue on this:

https://github.com/ivoa/ivoa-web/issues/90

JeremyMcCormick commented:
Instead of having instructions in contributing (which is full of other technical content), maybe we should have a dedicated `NEWSLETTER.md` for that information?

gmantele commented:
> FYI, there is an open issue on this:
> 
> #90
> 
> We would usually have this branch be called `iss90` but I'm okay with leaving it as is.

Sorry, I did not know about this issue (and I did not search for one....my bad). I linked this PR with this Issue.

gmantele commented:
> Instead of having instructions in `CONTRIBUTING.md` (which is full of other technical content), maybe we should have a dedicated `NEWSLETTER.md` for that information?

Ok. I do that.

molinaro-m commented:
Thanks for this setup.
It's important to contact the newsletter's editors to let them know about this and encourage them to use this template right away. I think the oncoming newsletter will be issue and reported on the current/old website, but having it already aligned with the new/rebuilt one would be great.

gmantele commented:
@molinaro-m , I let @JeremyMcCormick lead the coordination/communication with the newsletter's editors, unless any of you want me to speak for this specific point (in this case, please send me by email their email address).

@JeremyMcCormick , I have moved the instructions about the edition of newsletters in a separate MD file, as you've suggested. I have also added more information on how to edit the template. It is surely more technical, but I am sure it will be useful in the future ; few times already the newsletter layout/structure/content has evolved, so it may happen again.

JeremyMcCormick reviewed (approved):
I tested locally and this looks great - I approve.

Feel free to make minor adjustments/improvements before merging.

My only suggestion is that maybe we should add a Make target so a somewhat simpler command like `make newsletter` can be used to automatically use the installed hugo binary, and install it if necessary.

gmantele commented:
> My only significant request is that we have a Make target like `make newsletter [name]` which could be used to generate the page and automatically install/use the installed hugo binary.

Very good suggestion :+1: 

It is now done. Note that no parameter is required with this `make` command. The file name of the next newsletter can be easily automatically created: just increment the ID of the last newsletter. That's what this `make newsletter` does while printing instructions on how to continue (these instructions may be improved by the way).
